.github/workflows/create-alpha-release-branch.yaml

```diff
@@ -0,0 +1,80 @@
+name: Create Alpha Release Branch
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Alpha Release Branch Version'
+        required: false
+
+env:
+  artifactId: gaffer2
+  MAVEN_OPTS: -Dmaven.wagon.http.retryHandler.count=3 -Dmaven.wagon.httpconnectionManager.ttlSeconds=25
+
+jobs:
+  create-release-branch:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout v2-alpha
+      uses: actions/checkout@v2
+      with:
+        ref: v2-alpha
+        token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
+        fetch-depth: 0
+
+    - name: Set version from input
+      if: ${{ github.event.inputs.version }}
+      run: echo "RELEASE_VERSION=$(echo ${{ github.event.inputs.version }} | sed 's/^v//')" >> $GITHUB_ENV
+
+    - name: Get latest tag
+      if: ${{ !github.event.inputs.version }}
+      uses: actions-ecosystem/action-get-latest-tag@v1
+      id: get-latest-tag
+
+    - name: Format latest tag
+      if: ${{ !github.event.inputs.version }}
+      run: echo "CURRENT_VERSION=$(echo ${{ steps.get-latest-tag.outputs.tag }} | sed 's/^gaffer2-//')" >> $GITHUB_ENV
+
+    - name: Bump latest tag variable
+      if: ${{ !github.event.inputs.version }}
+      uses: actions-ecosystem/action-bump-semver@v1
+      id: bump-semver
+      with:
+        current_version: ${{ env.CURRENT_VERSION }}
+        level: prerelease
+
+    - name: Set version automatically
+      if: ${{ !github.event.inputs.version }}
+      run: echo "RELEASE_VERSION=$(echo ${{ steps.bump-semver.outputs.new_version }} )" >> $GITHUB_ENV
+
+    - name: Verify version regex
+      uses: actions-ecosystem/action-regex-match@v2
+      id: regex-match
+      with:
+        text: ${{ env.RELEASE_VERSION }}
+        # Ensure version starts in 2
+        regex: '^(2)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
```

Comment:
A bit messy and lets you enter in strange version numbers such as `2.0.0-a0-+0-` (see [https://regex101.com/](url)).

Breaking it up into smaller reusable parts would help if that is possible

Reply:
I agree it is weird but it is the official suggested regex on semver.org: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

Reply:
I also think this is weird and overly complex. Most of that complexity seems to be to prevent versions with leading zeros, like `01.023.03`. It also has capturing groups which we don't need. We could use something like `^\d+\.\d+\.\d+(?:-\w+)?$` if we just want a simple check. Maybe leave as it and add that link to explain why it's weird?

Reply:
Yeah perhaps a link. If there is a suggested would be a shame not to use it.